src/coreclr/jit/codegenwasm.cpp

```diff
@@ -105,6 +105,39 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
     }
 }
 
+//------------------------------------------------------------------------
+// PackOperAndType: Pack a genTreeOps and var_types into a uint32_t
+//
+// Arguments:
+//    oper - a genTreeOps to pack
+//    type - a var_types to pack
+//
+// Return Value:
+//    oper and type packed into an integer that can be used as a switch value/case
+//
+static constexpr uint32_t PackOperAndType(genTreeOps oper, var_types type)
+{
+    if (type == TYP_BYREF)
+    {
+        type = TYP_I_IMPL;
+    }
+    return ((uint32_t)oper << (ConstLog2<GT_COUNT>::value + 1)) | ((uint32_t)type);
```

Comment:
```suggestion
    static_assert(GT_COUNT > TYP_COUNT);
    return ((uint32_t)oper << (ConstLog2<GT_COUNT>::value + 1)) | ((uint32_t)type);
```
Otherwise we may hypothetically mix up the bits.

Reply:
Yeah, better safe than sorry.